mpitests/test_apply.py

```diff
@@ -439,32 +485,527 @@ def test_dev_reset(self, tol, operation, par, Wires):
+
+
+@pytest.mark.parametrize(
+    "returns",
+    [
+        (qml.PauliX(0),),
+        (qml.PauliY(0),),
+        (qml.PauliZ(0),),
+        (qml.PauliX(1),),
+        (qml.PauliY(1),),
+        (qml.PauliZ(1),),
+        (qml.PauliX(2),),
+        (qml.PauliY(2),),
+        (qml.PauliZ(2),),
+        (qml.PauliX(3),),
+        (qml.PauliY(3),),
+        (qml.PauliZ(3),),
+        (qml.PauliX(0), qml.PauliY(1)),
+        (
+            qml.PauliZ(0),
+            qml.PauliX(1),
+            qml.PauliY(2),
+        ),
+        (
+            qml.PauliY(0),
+            qml.PauliZ(1),
+            qml.PauliY(3),
+        ),
+        (qml.PauliZ(0) @ qml.PauliY(3),),
+        (qml.Hadamard(2),),
+        (qml.Hadamard(3) @ qml.PauliZ(2),),
+        (qml.PauliX(0) @ qml.PauliY(3),),
+        (qml.PauliY(0) @ qml.PauliY(2) @ qml.PauliY(3),),
+        (qml.PauliZ(0) @ qml.PauliZ(1) @ qml.PauliZ(2),),
+        (0.5 * qml.PauliZ(0) @ qml.PauliZ(2),),
+    ],
+)
+def test_integration(returns):
+    """Integration tests that compare to default.qubit for a large circuit containing parametrized
+    operations"""
+    num_wires = numQubits
+    comm = MPI.COMM_WORLD
+    dev_default = qml.device("default.qubit", wires=range(num_wires))
+    dev_gpu = qml.device("lightning.gpu", wires=num_wires, mpi=True, c_dtype=np.complex128)
+
+    def circuit(params):
+        circuit_ansatz(params, wires=range(num_wires))
+        return [qml.expval(r) for r in returns]
+
+    n_params = 30
+    np.random.seed(1337)
+    params = np.random.rand(n_params)
+
+    qnode_gpu = qml.QNode(circuit, dev_gpu, diff_method="parameter-shift")
+    qnode_default = qml.QNode(circuit, dev_default, diff_method="parameter-shift")
+
+    def convert_to_array_gpu(params):
+        return np.array(qnode_gpu(params))
+
+    def convert_to_array_default(params):
+        return np.array(qnode_default(params))
+
+    j_gpu = qml.jacobian(convert_to_array_gpu)(params)
+    j_default = qml.jacobian(convert_to_array_default)(params)
+
+    assert np.allclose(j_gpu, j_default, atol=1e-7)
+
+
+custom_wires = ["alice", 3.14, -1, 0, "bob", "l", "m", "n"]
+
+
+@pytest.mark.parametrize(
+    "returns",
+    [
+        qml.PauliZ(custom_wires[0]),
+        qml.PauliX(custom_wires[2]),
+        qml.PauliZ(custom_wires[0]) @ qml.PauliY(custom_wires[3]),
+        qml.Hadamard(custom_wires[2]),
+        qml.Hadamard(custom_wires[3]) @ qml.PauliZ(custom_wires[2]),
+        qml.PauliX(custom_wires[0]) @ qml.PauliY(custom_wires[3]),
+        qml.PauliY(custom_wires[0]) @ qml.PauliY(custom_wires[2]) @ qml.PauliY(custom_wires[3]),
+    ],
+)
+def test_integration_custom_wires(returns):
+    """Integration tests that compare to default.qubit for a large circuit containing parametrized
+    operations and when using custom wire labels"""
+    comm = MPI.COMM_WORLD
+    dev_lightning = qml.device("lightning.qubit", wires=custom_wires)
```

Comment:
In hindsight, I think comparing to `default.qubit` here makes more sense, as we tend to follow the same structure across the lightning devices.

Reply:
Sure! Done